### PR TITLE
Implement Add Capacity Modal using PF

### PR DESCRIPTION
**Pull request by rexagod**
Fixes: https://issues.redhat.com/browse/RHSTOR-1108

#### Before
image

#### Now
image

### CODE REVIEW

**Comment by rexagod** [comment]
The icons seem different, fixing those.

**Comment by rexagod** [comment]
Seems to pass locally. Looking into this.

image

**Comment by afreen23** [comment]
Update the i18n files: run `yarn i18n`  and commit the chnages

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 1]
```diff
@@ -129,70 +137,107 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
-        <Trans t={t} ns="ceph-storage-plugin" values={{ name }}>
-          Adding capacity for <strong>{{ name }}</strong>, may increase your expenses.
-        </Trans>
-        <div className="ceph-add-capacity__modal">
-          <div
-            className={classNames('ceph-add-capacity__sc-dropdown', {
-              'ceph-add-capacity__sc-dropdown--margin': !isNoProvionerSC,
-            })}
+    <Flex direction={{ default: 'column' }}>
+      <Form className="pf-u-background-color-100 pf-u-pt-xl pf-u-pl-xl">
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:141]: Why do we need to set backgorund color, we should be using the default only.

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 2]
```diff
@@ -129,70 +137,107 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
+        <FlexItem>
+          <Title headingLevel="h4" size="xl">
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:143]: We can use `ModalTitle`, `ModalBody` and `ModalFooter` they have predefined css and are set as per conventions.

**Review by afreen23 — CHANGES_REQUESTED** [review]
By changing to Pf forms , we want to change to Form elements provided by Pf like FormGroup/TextContent , etc.
We still want to keep using Modal Title, Body and Footer.

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 1 above)
Comment (rexagod) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: This went transparent so I had to include this class. Fixed now.

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 3]
```diff
@@ -135,56 +137,66 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
+        <FormGroup
+          className="pf-u-pt-md pf-u-pb-sm"
+          id="add-cap-sc-dropdown__FormGroup"
+          fieldId="add-cap-sc-dropdown__FormGroup"
+          isRequired
+        >
+          <Title className="pf-u-pb-sm" headingLevel="h5">
+            {t('ceph-storage-plugin~Storage Class')}&nbsp;&nbsp;
+            <Tooltip position="top" content={<>{storageClassTooltip(t)}</>}>
+              <QuestionCircleIcon className="pf-u-secondary-color-100" />
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:149]: Form group has a `label` prop which can take a react element as well as a `labelIcon` to support the icon.
Refer this for more help: https://www.patternfly.org/v4/components/form
I would assume we can leverage both of these props here.

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 4]
```diff
@@ -135,56 +137,66 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
+            </Tooltip>
+          </Title>
+          <div className="ceph-add-capacity__sc-dropdown">
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:152]: Add the id used in `FormField` here , it is for semantics and accessibility

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 5]
```diff
@@ -135,56 +137,66 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
+            <FormGroup
+              className="pf-u-py-sm"
+              fieldId="requestSize__FormGroup"
+              id="requestSize__FormGroup"
+            >
+              <Title className="pf-u-pb-sm" headingLevel="h5">
+                {t('ceph-storage-plugin~Raw Capacity')}&nbsp;&nbsp;
+                <Tooltip position="top" content={<>{requestedCapacityTooltip(t)}</>}>
+                  <QuestionCircleIcon className="pf-u-secondary-color-100" />
+                </Tooltip>
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:173]: here as well , make sure to use `label` and `labelIcon` props on `FormGroup`

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 6]
```diff
@@ -135,56 +137,66 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
+              </Title>
+              <TextInput
+                isDisabled
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:176]: Whenever you add a child to `FormGroup` , make sure to add the the `id` to chld with value being same as `fieldId`.
It connects them semantically and hence a`required` prop.

**Review by afreen23 — COMMENTED** [review]
I can see we are still using `<form` instead of Pf `Form` tags.
Refer this https://www.patternfly.org/v4/components/form

**Comment by rexagod** [comment]
> I can see we are still using <form instead of Pf Form tags.
> Refer this https://www.patternfly.org/v4/components/form

I was actually using `Form` in the previous commits (https://github.com/openshift/console/pull/8178/commits/621afeca3205be8302e48c64d8212fe9ea125d24) but I was unable to find `onSubmit` prop on the `Form` component inorder to make it work with `ModalFooter`, so I resorted to using the `form` component and removed [how I handled the submit event earlier](https://github.com/openshift/console/pull/8178/commits/621afeca3205be8302e48c64d8212fe9ea125d24#diff-75a10ade40aa0f0328dbd77cc35d6db8360ca15231d1aa142309a9d5465d1269L219) so to make `form` compliant with `ModalFooter`.

I'll look into this a bit and get back to you.

Thank you for the detailed review!

**Comment by rexagod** [comment]
> I can see we are still using <form instead of Pf Form tags.
> Refer this https://www.patternfly.org/v4/components/form

Apologies for the confusion, I was actually using the `Form` component till the second last commit but changed it to `form` in the latest one because for some reason my ts-lint complained about `onSubmit` not being present on `Form`. I checked this again today and it seems to detect it and works fine now.

**Comment by rexagod** [comment]
As you can see below, `pf-c-form`, that comes in with the `Form` component, enforces the `grid` as the display property, cause the form to break-up in parts. I used [`pf-u-display-block`](https://github.com/openshift/console/pull/8178/commits/00a1ad7bdd8d3f52a6cc643dadff4d192247cefe#diff-75a10ade40aa0f0328dbd77cc35d6db8360ca15231d1aa142309a9d5465d1269R136) to suppress this. Please let me know if you think any change is required and I'll incorporate them.

2021-03-08_02:01_1784x462

**Comment by rexagod** [comment]
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_console/8178/pull-ci-openshift-console-master-e2e-gcp-console/1371812009359708160/artifacts/e2e-gcp-console/test/artifacts/gui_test_screenshots/cypress/screenshots/catalog-source-details.spec.ts/

**Comment by afreen23** [comment]
/lgtm cancel
squash the commits

**Comment by rexagod** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_console/8178/pull-ci-openshift-console-master-ceph-storage-plugin/1374386551991046144#1:build-log.txt:127

**Comment by rexagod** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_console/8178/pull-ci-openshift-console-master-ceph-storage-plugin/1374391016441778176#1:build-log.txt%3A127

File: frontend/packages/ceph-storage-plugin/locales/en/ceph-storage-plugin.json [hunk 7]
```diff
@@ -343,7 +343,7 @@
   "Storage Cluster exists": "Storage Cluster exists",
   "Back to operator page": "Back to operator page",
   "Go to cluster page": "Go to cluster page",
-  "clusterExistText": "A storage cluster <1>{{clusterName}}</1> is already created.<3></3>You cannot create another storage cluster.",
+  "clusterExistText": "A storage cluster <1>{{clusterName}}</1> already exists.<3></3>You cannot create another storage cluster.",
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/locales/en/ceph-storage-plugin.json:346]: Why this change ?

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/locales/en/ceph-storage-plugin.json (hunk 7 above)
Comment (rexagod) [reply frontend/packages/ceph-storage-plugin/locales/en/ceph-storage-plugin.json:346]: I think this might have cropped up when I ran `yarn run i18n` on this branch a while ago. Although I didn't make this change, I thought this was something the aforementioned command did, not sure why.

Should I go ahead and revert it?

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/locales/en/ceph-storage-plugin.json:346]: yarn run i18n would not add this. Possibly some rebase issue.
Please rebase your PR with latest master, it will vanish.
